src/web_app_skeleton/docker/dev_entrypoint.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# This is the entrypoint script used for development docker workflows. By
+# default it will:
+#  - Install shards.
+#  - Run migrations.
+#  - Start the dev server.
+# It also accepts any commands to be run instead.
+
+
+warnfail () {
+  echo "$@" >&2
+  exit 1
+}
+
+case ${1:-} in
+  "") # If no arguments are provided, start lucky dev server.
+    ;;
+
+  *) # If any arguments are provided, execute them instead.
+    exec "$@"
+esac
+
+echo "Installing npm packages..."
+npm install
+
+if ! shards check ; then
+  echo "Installing shards..."
+  shards install
```

Comment:
(this does also mean you don't need your build tools or `node_modules` in the docker image, making them MUCH smaller!)

Reply:
I mounted this stuff at boot time rather than at build time because I want to be able to capture the output of these commands in the docker volume so that a container rebuild doesn't require re-downloading the modules/shards. In a release image I would absolutely put them in the build phase.